### ISSUE TITLE
Set password when creating a channel

### ISSUE THREAD

**Issue by eliocamp** [thelounge/thelounge#4228]
### Feature Description

When joining a channel, there's an optional password field. If the channel already exists and needs a password, it is used. But if the channel does not exists and it's created, this is ignored.
This behaviour, I think, is a bit counter-intuitive. I would naively expect that if I'm creating a channel and typing a password, the resulting channel would be password-protected. It would it be more great if  The Lounge could set up a password when creating a new channel.

**Comment by brunnre8** [comment]
Hm the reason for this is that it's actually two different operations.

/join simply joins, if that channel doesn't exist the server creates it and sets the default modes.

If you then want to set a password you need to do eg `MODE #42 +k oulu` same for any other channel mode changes like invite only or a banmask or whatever.

TL actually just passes the /join command through without doing anything with it. The server then decides how to handle it.

**Comment by eliocamp** [comment]
Yep, I figured that this might be the case. The question is if it could be possible for The Lounge to detect the result and send the proper command to set the password if needed.

**Comment by brunnre8** [comment]
I'm not actually sure of we can race free detect whether we made a new channel from the servers perspective or got oped some other way...
And you certainly don't want to suddenly try to set a password just because you typoed and have a trailing letter when typing /join from TL.

After sending /join the server will reply with the chan modes and the users which are currently in the chan, but I don't think that's indicated in any way special

**Comment by eliocamp** [comment]
Maybe if the user is the only user on the channel and it's operator then assume that the channel was created and send the /mode commands to add the password? Does that make any sens?

**Comment by brunnre8** [comment]
I think this should better be done on the ircd itself, not client side

**Comment by eliocamp** [comment]
Perhaps. In that case, is there something that could be changed in the UI to make it clear that the password field won't apply to a newly created channel?

**Comment by brunnre8** [comment]
It does apply... in the same sense that it applies if you stick a key that doesn't actually work into the keyslot of a door that's not actually locked and then turn the knob and the door opens.

It's a password for entering the secret lair and not a command that sets one.

And actually, we don't have it documented at all that you can do that,

```
/join channel
Join a channel.
```

that's literally the full help text TL gives you in the help section (which we should patch I guess)

**Comment by eliocamp** [comment]
To be clear, I'm not talking about the /join command, I'm talking about the channel + password text input that appears when clicking on the "+" icon next to a network. Since those fields are used both for joining an existing channel and for creating a new channel, I think it's confusing for people who are not aware of the lower-level commands that TL is executing. (I'm not speculating, BTW, we actually tested The Lounge and some people were confused by this).

**Comment by brunnre8** [comment]
heh, I never actually pushed that button...

Good, so the confusion was that people expected the channel to then have a password? Just because they joined it as the first user?

What gave them that notion? All other channel modes aren't set this way either. And I mean things like invite only, quiet etc.

Not sure how we can make this more obvious actually... we don't particularly have a lot of space where the widget pops up for some exemplary text.

Do you have any idea what wording would be more obvious?

**Comment by eliocamp** [comment]
> What gave them that notion? 

I don't think is at all unreasonable to expect that if you are creating a channel and that the "name" field populates the name of the channel, then the "password" field will also be reflected in the password of the channel. People that don't use IRC don't necessarily think in terms of /join or channel modes. They are thinking "I'm creating a channel with this name and this password". They might not even know what "modes" are. 

> Do you have any idea what wording would be more obvious?

Sadly, not at the moment. Perhaps something alon gthe lines of "Password (only if requested by channel)" ?

**Comment by eliocamp** [comment]
Here's what The Lounge shows now when connecting to a network. Can't it show the same when joining a channel?

![screenshot  of connection screen. Shows an active tickbox that reads "I have a password" and an input text with label "Password" and "server password (optional)" as placeholder](https://user-images.githubusercontent.com/8617595/119046706-6756b980-b993-11eb-94e8-d5aa649454a2.png)

**Comment by brunnre8** [comment]
where's that screen from?
I'm probably not running the same version as you, my version doesn't have the checkbox

![Screenshot_20210521-070225](https://user-images.githubusercontent.com/16165454/119084733-9fe1ac00-ba02-11eb-8dd5-74f39aaa10f8.png)

And it should not be a checkbox anyhow, just a label.

**Comment by eliocamp** [comment]
It's from the demo version: https://demo.thelounge.chat/
